alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/model/TxEntities.java

```diff
@@ -17,20 +17,31 @@
 
 package org.apache.servicecomb.pack.alpha.fsm.model;
 
-import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
 import org.apache.servicecomb.pack.alpha.core.fsm.TxState;
 
 public class TxEntities {
 
-  private ConcurrentSkipListMap<String, TxEntity> entities = new ConcurrentSkipListMap<>();
+  private ConcurrentHashMap<String, TxEntity> entities = new ConcurrentHashMap<>();
 
   public void forEach(BiConsumer<String, TxEntity> action) {
-    entities.forEach(action);
+    entities.entrySet().stream()
+        .sorted((v1, v2) -> v1.getValue().getBeginTime().compareTo(v2.getValue().getBeginTime()))
```

Comment:
I think this store action will introduce some performance issue, if there are lot of txEntity need to be visited.

Reply:
I found that there is no multi-threaded access to the TxEntities class, I think I can use LinkedHashMap to save sub-transaction objects. By default, LinkedHashMap maintains the insertion order

Reply:
done